diff --git i/GigaLearnCPP/src/private/GigaLearnCPP/Util/Models.h w/GigaLearnCPP/src/private/GigaLearnCPP/Util/Models.h
index 644d2ca..e5ab48b 100644
--- i/GigaLearnCPP/src/private/GigaLearnCPP/Util/Models.h
+++ w/GigaLearnCPP/src/private/GigaLearnCPP/Util/Models.h
@@ -190,8 +190,14 @@ namespace GGL {
 				model->Load(folder, allowNotExist, loadOptims);
 		}
 
-		class ModelIterator : public std::iterator<std::forward_iterator_tag, typename Model*> {
+		class ModelIterator {
 		public:
+			using iterator_category = std::forward_iterator_tag;
+			using value_type = Model*;
+			using difference_type = std::ptrdiff_t;
+			using pointer = Model**;
+			using reference = Model*&;
+
 			using MapItr = std::map<std::string, Model*>::iterator;
 			MapItr _mapItr;
 
@@ -202,7 +208,7 @@ namespace GGL {
 			bool operator==(const ModelIterator& other) const { return _mapItr == other._mapItr; }
 			bool operator!=(const ModelIterator& other) const { return _mapItr != other._mapItr; }
 
-			typename Model*& operator*() const { return _mapItr->second; }
+			Model*& operator*() const { return _mapItr->second; }
 		};
 
 		ModelIterator begin() {
